packages/extension-mathematics/src/extensions/BlockMath.ts

```diff
@@ -0,0 +1,210 @@
+import { InputRule, mergeAttributes, Node } from '@tiptap/core'
+import type { Node as PMNode } from '@tiptap/pm/model'
+import katex from 'katex'
+
+export type BlockMathOptions = {
+  onClick?: (node: PMNode, pos: number) => void
+}
+
+declare module '@tiptap/core' {
+  interface Commands<ReturnType> {
+    blockMath: {
+      /**
+       * Set block math node with LaTeX string.
+       * @param options - Options for setting block math.
+       * @returns ReturnType
+       */
+      setBlockMath: (options: { latex: string; pos?: number }) => ReturnType
+
+      /**
+       * Unset block math node.
+       * @returns ReturnType
+       */
+      unsetBlockMath: (options?: { pos?: number }) => ReturnType
+
+      /**
+       * Update block math node with optional LaTeX string.
+       * @param options - Options for updating block math.
+       * @returns ReturnType
+       */
+      updateBlockMath: (options?: { latex?: string; pos?: number }) => ReturnType
+    }
+  }
+}
+
+/**
+ * BlockMath is a Tiptap extension for rendering block mathematical expressions using KaTeX.
+ * It allows users to insert LaTeX formatted math expressions block within text.
+ * It supports rendering, input rules for LaTeX syntax, and click handling for interaction.
+ *
+ * @example
+ * ```javascript
+ * import { BlockMath } from 'your-extension-path'
+ * import { Editor } from '@tiptap/core'
+ *
+ * const editor = new Editor({
+ *   extensions: [
+ *     BlockMath.configure({
+ *       onClick: (node, pos) => {
+ *         console.log('Block math clicked:', node.attrs.latex, 'at position:', pos)
+ *       },
+ *     }),
+ *   ],
+ * })
+ */
+export const BlockMath = Node.create({
+  name: 'blockMath',
+
+  group: 'block',
+
+  atom: true,
+
+  addOptions() {
+    return {
+      onClick: undefined,
+    }
+  },
+
+  addAttributes() {
+    return {
+      latex: {
+        default: '',
+        parseHTML: element => element.getAttribute('data-latex'),
+        renderHTML: attributes => {
+          return {
+            'data-latex': attributes.latex,
+          }
+        },
+      },
+    }
+  },
+
+  addCommands() {
+    return {
+      setBlockMath:
```

Comment:
This behavior seems inconsistent to me with previous `set` and `unset` commands. Shouldn't this wrap a selected text as a BlockMath? Would a different name make more sense? Or do so if no `latex` is passed and a selection is given?